### PR TITLE
add support for optional callback

### LINKED ISSUE #60: optional callback
It would be nice if time-grunt accepted an optional callback function that gets passed the timing stats. For my case I'm wanting to build a graph of my grunt tasks' speed over time. Would you accept a PR for this?

### PR DESCRIPTION
**Pull request by geddski**
closes #60

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -146,6 +147,14 @@ module.exports = function (grunt) {
 		// `grunt.log.header` should be unhooked above, but in some cases it's not
 		log('\n\n' + chalk.underline('Execution Time') + chalk.gray(' (' + startTimePretty + ')'));
 		log(formatTable(tableData) + '\n');
-		process.exit(exitCode);
+		
+		if (module.exports.callback){
+			module.exports.callback(tableData, function(){
```
Comment (kevva) [inline index.js]: `function () {`

File: index.js [hunk 2]
```diff
@@ -146,6 +147,14 @@ module.exports = function (grunt) {
 		// `grunt.log.header` should be unhooked above, but in some cases it's not
 		log('\n\n' + chalk.underline('Execution Time') + chalk.gray(' (' + startTimePretty + ')'));
 		log(formatTable(tableData) + '\n');
-		process.exit(exitCode);
+		
+		if (module.exports.callback){
```
Comment (kevva) [inline index.js]: `if (module.exports.callback) {`

File: index.js [hunk 3]
```diff
@@ -13,7 +13,8 @@ function log(str) {
 	write(str + '\n', 'utf8');
 }
 
-module.exports = function (grunt) {
+module.exports = function (grunt, callback) {
+	module.exports.callback = callback;
```
Comment (kevva) [inline index.js]: Why does this need exporting?

File: index.js [hunk 4]
```diff
@@ -13,7 +13,8 @@ function log(str) {
 	write(str + '\n', 'utf8');
 }
 
-module.exports = function (grunt) {
+module.exports = function (grunt, callback) {
```
Comment (kevva) [inline index.js]: `function (grunt, cb)`

File: index.js [hunk 5]
```diff
@@ -146,6 +147,14 @@ module.exports = function (grunt) {
...
 		log('\n\n' + chalk.underline('Execution Time') + chalk.gray(' (' + startTimePretty + ')'));
 		log(formatTable(tableData) + '\n');
-		process.exit(exitCode);
+		
+		if (module.exports.callback){
+			module.exports.callback(tableData, function(){
+				process.exit(exitCode);
+			});
+		}
+		else {
```
Comment (kevva) [inline index.js]: No need for `if/else`. Just put a `return` after the call to the callback in the `if` statement.

File: readme.md [hunk 6]
```diff
@@ -21,6 +21,18 @@ module.exports = function (grunt) {
 }
 ```
 
+## Optional Callback
+If you want to collect the timing stats for your own use, pass a callback to time-grunt:
```
Comment (kevva) [inline readme.md]: Add a blank line between the heading and the text.

File: readme.md [hunk 7]
```diff
@@ -21,6 +21,18 @@ module.exports = function (grunt) {
...
+
+```js
+require('time-grunt')(grunt, function(stats, done){
```
Comment (kevva) [inline readme.md]: `function(stats, done) {`

**Comment by geddski** [comment]
@kevva thx for the review.
@sindresorhus this is ready when you are :) Would appreciate a version bump after the merge. Thanks again! Great little tool this is.

File: index.js [hunk 8]
```diff
@@ -146,6 +146,13 @@ module.exports = function (grunt) {
 		// `grunt.log.header` should be unhooked above, but in some cases it's not
 		log('\n\n' + chalk.underline('Execution Time') + chalk.gray(' (' + startTimePretty + ')'));
 		log(formatTable(tableData) + '\n');
+		
+		if (callback) {
+			callback(tableData, function() {
```
Comment (arthurvr) [inline index.js]: `function () {`

File: readme.md [hunk 9]
```diff
@@ -21,6 +21,19 @@ module.exports = function (grunt) {
 }
 ```
 
+## Optional Callback
+
+If you want to collect the timing stats for your own use, pass a callback to time-grunt:
+
+```js
+require('time-grunt')(grunt, function(stats, done) {
```
Comment (arthurvr) [inline readme.md]: `function (stats, done) {`

**Comment by sindresorhus** [comment]
Looks ok when the remaining inline comments are addressed.

**Comment by geddski** [comment]
resolved. Thanks!

**Comment by cletusw** [comment]
@geddski It looks like this doesn't wait for any asynchronous tasks to finish before killing the process.
```javascript
require('time-grunt')(grunt, function(stats, done){
  // throw 'synchronous 1'; // Uncomment to show that this *does* get executed
  setTimeout(function () {
    throw 'asynchronous'; // This, however, will never run
    done();
  }, 100);
  // throw 'synchronous 2'; // Uncomment to show that this *does* get executed
});
```

**Pull request merged**